ghcide/src/Development/IDE/Core/Compile.hs

```diff
@@ -1000,23 +1000,25 @@ getDocsBatch
   -> [Name]
   -> IO [Either String (Maybe HsDocString, Map.Map Int HsDocString)]
 getDocsBatch hsc_env _mod _names = do
-    ((_warns,errs), res) <- initTc hsc_env HsSrcFile False _mod fakeSpan $ forM _names $ \name ->
+    ((_warns,errs), res) <- initTc hsc_env HsSrcFile False _mod fakeSpan $ traverse findNameInfo _names
+    case res of
+        Just x  -> return $ map (first $ T.unpack . showGhc) x
+        Nothing -> throwErrors errs
+  where
+    throwErrors = liftIO . throwIO . mkSrcErr
+
+    findNameInfo :: Name -> IOEnv (Env TcGblEnv TcLclEnv) (Either GetDocsFailure (Maybe HsDocString, Map.Map Int HsDocString))
+    findNameInfo name =
         case nameModule_maybe name of
             Nothing -> return (Left $ NameHasNoModule name)
             Just mod -> do
              ModIface { mi_doc_hdr = mb_doc_hdr
                       , mi_decl_docs = DeclDocMap dmap
                       , mi_arg_docs = ArgDocMap amap
                       } <- loadModuleInterface "getModuleInterface" mod
-             if isNothing mb_doc_hdr && Map.null dmap && Map.null amap
-               then pure (Left (NoDocsInIface mod $ compiled name))
-               else pure (Right ( Map.lookup name dmap
-                                , Map.findWithDefault Map.empty name amap))
-    case res of
-        Just x  -> return $ map (first $ T.unpack . showGhc) x
-        Nothing -> throwErrors errs
-  where
-    throwErrors = liftIO . throwIO . mkSrcErr
+             pure $ if isNothing mb_doc_hdr && Map.null dmap && Map.null amap
+               then Left (NoDocsInIface mod $ compiled name)
+               else Right (Map.lookup name dmap, Map.findWithDefault mempty name amap)
```

Comment:
Here I just form a local function (& factor-out `pure`). And the `getDocsBatch` scope code goes up.